## You have a total of 48 hours to complete these questions. Each hour late in excess of 48 hours will penalize your total score by 5 points. The questions are in the easy-to-hard order.

- How is sequencing saturation calculated in CellRanger? (10)

- Explain the causes of high level of mitochondrial gene expression in scRNA-Seq data. (10)

- How to modify the STAR alignment parameters in CellRanger? How to modify them in CeleScope? (10)

- What is the difference between Seurat data integration and merging? When will you use `IntegrateData` instead of `merge`? (10)

- Install SingleR through Conda and use SingleR to annotate [pbmc3k](https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz) data. (10)

- Using pbmc3k data, write a script to:
    1. Plot the average UMI per gene per cell on t-SNE plot. (10)
    2. Generate an expression heatmap for the top 10 markers(10 each, total 30 genes) of Naive CD4+ T, CD8+ T and NK cells. (10)

- 10X V2 chemistry has 10-bp long UMI but 12-bp in V3. What are the advantages and disadvantages of longer UMI? (10)

- Write a script to subset 100 cells with the highest UMI count from pbmc3k data and save the matrix of these 100 cells in 10X format. (20)

## Bonus question

- Write a script to calculate the sequencing saturation of the CeleScope rna test data(https://github.com/singleron-RD/celescope_test_script/tree/master/rna). (20)